python/ray/_private/accelerators/__init__.py

```diff
@@ -0,0 +1,40 @@
+from typing import Set
+
+from ray._private.accelerators.accelerator import Accelerator
+from ray._private.accelerators.nvidia_gpu import NvidiaGPUAccelerator
+from ray._private.accelerators.tpu import TPUAccelerator
+from ray._private.accelerators.neuron import NeuronAccelerator
+
+
+def get_all_accelerators() -> Set[Accelerator]:
+    """Get all accelerators supported by Ray."""
+    return {
+        NvidiaGPUAccelerator,
+        TPUAccelerator,
+        NeuronAccelerator,
+    }
+
+
+def get_all_accelerator_resource_names() -> Set[str]:
```

Comment:
Why don't we just use enums here instead of names directly?

Reply:
What benefits do enum provide here?

Reply:
I think the top-level API makes sense to accept string, but for internal functions, it's cleaner to pass enum around (otherwise, we should rely on some implicit assumption the input is always valid, or we should do validation everywhere).